src/test/java/org/jnode/fs/ntfs/NTFSCompressedDataRunTriage4924Test.java

```diff
@@ -0,0 +1,191 @@
+
+    @Test
+    public void testDecompression_read_chunk_missing_previous_chunk() throws IOException {
+        // Arrange
+        byte[] data = FileSystemTestUtils.readFileToByteArray("org/jnode/fs/ntfs/triage-4924-E01.compressed.bin");
+
+        // Deliberately skip the previous chunk.
+        // the new chunk starts from 787.
+        byte[] compressed = new byte[data.length - 787];
+        System.arraycopy(data, 787, compressed, 0, data.length - 787);
+        byte[] uncompressed = new byte[0x20000];
+
+        // Act
+        try {
+            //the offset goes to a position before the first bit of the chunk (so it goes to the chunk before the current one)
+            CompressedDataRun.unCompressUnit(compressed, uncompressed);
+        } catch (Exception e) {
+            assertThat(e instanceof ArrayIndexOutOfBoundsException, is(true));
+            return;
+        }
+
+        // should not go here
+        assertThat(true, is(false));
+    }
+
+    @Test
+    public void testDecompression_E01() throws IOException {
+        // Arrange
+        byte[] compressed = FileSystemTestUtils.readFileToByteArray("org/jnode/fs/ntfs/triage-4924-E01.compressed.bin");
+
+        byte[] uncompressed = new byte[0x20000];
+
+        // Act
+        CompressedDataRun.unCompressUnit(compressed, uncompressed);
+
+        // Assert
+        String uncompressedContent = new String(uncompressed, StandardCharsets.US_ASCII);
+        assertThat(uncompressedContent, containsString(
+                "PublicKey=002400000480000094000000060200000024000052534131000400000100010007d1fa57c4aed9f0a32e84aa0faefd0de9e8fd6aec8f87fb03766c834c99921eb2" +
+                        "3be79ad9d5dcc1dd9ad236132102900b723cf980957fc4e177108fc607774f29e8320e92ea05ece4e821c0a5efe8f1645c4c0c93c1ab99285d622caa652c1dfad63d" +
+                        "745d6f2de5f17e5eaf0fc4963d261c8a12436518206dc093344d5ad293")); // Near the end of the chunk
+    }
+
+    @Test
+    public void testDecompression_E01_offsetBeforeStart() throws IOException {
+        // Arrange
+        byte[] compressed = FileSystemTestUtils.readFileToByteArray("org/jnode/fs/ntfs/offset-before-start.compressed.bin");
+
+        byte[] uncompressed = new byte[0x20000];
+
+        // Act
+        // there is an error, but we ignore it for now to get more meaningful data out.
+        CompressedDataRun.unCompressUnit(compressed, uncompressed);
+
+        // Assert
+        byte[] expectedByteArray = {
+                40, 126, 9, 0, 6, 114, 32, -79, 0, 112, 25, -115, 28,
+                0, 0, 1, 37, 22, 17, 8, 111, 56, 19, 0, 6, 114, -15,
+                31, 0, 112, 111, 49, 1, 0, 10, -94, 37, 23, 17, 8,
+                111, 53, 19, 0, 6, 17, 8, 111, 56, 19, 0, 6, 111,
+                -95, 0, 0, 10, 126, 112, 0, 0, 10, 111, -26, 2, 0,
+                10, -94, 37, 24, 9, 123, 4, 8, 0, 4, 111, 53, 19,
+                0, 6, 9, 123, 4, 8, 0, 4, 111, 56, 19, 0, 6, 111,
+                -95, 0, 0, 10, 126, 112, 0
```

Comment:
Maybe we can write the expected data into an "expected" file. E.g. `"org/jnode/fs/ntfs/offset-before-start.compressed_expected.bin"`